api/envoy/config/core/v3/protocol.proto

```diff
@@ -249,11 +249,9 @@ message HttpProtocolOptions {
   google.protobuf.Duration idle_timeout = 1;
 
   // The maximum duration of a connection. The duration is defined as a period since a connection
-  // was established. If not set, there is no max duration. When max_connection_duration is reached
-  // and if there are no active streams, the connection will be closed. If the connection is a
-  // downstream connection and there are any active streams, the drain sequence will kick-in,
-  // and the connection will be force-closed after the drain period. See :ref:`drain_timeout
-  // <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.drain_timeout>`.
+  // was established. If not set, there is no max duration. When max_connection_duration is reached,
+  // the connection will be closed if there are no active streams or the drain sequence will kick-in.
```

Comment:
I have always seen drain timeout kicking in after max-connection-duration(MCD) is reached on an existing connection. Do you see any factual error (based on your tests) with changing stmt to "**When max_connection_duration is reached, the connection is closed after drain timeout period if there are no active streams**". This essentially means, all connections wait for drain timeout after MCD ends and then try to close the connection if there are no active streams.